### PR TITLE
Maintenance updates

### PR DESCRIPTION
**Pull request by blink1073**
- Switch to hatch build backend
- Add test and CI setup

### CODE REVIEW

**Comment by blink1073** [comment]
Looks like we can't run any CI until this is merged

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+    "traitlets>5.3",
+]
+
+[project.optional-dependencies]
+test = [
+  "pytest",
+]
+
+[project.urls]
+Homepage = "https://github.com/martinRenou/comm"
```
Comment (martinRenou) [inline pyproject.toml:30]:
```suggestion
Homepage = "https://github.com/ipython/comm"
```

**Review by martinRenou — COMMENTED** [review]
Thank your for doing this!

**Comment by bollwyvl** [comment]
Also: always be classifying: framework::ipython, ::jupyter, etc,

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+requires-python = ">=3.6"
+authors = [
+    { name = "Jupyter contributors" },
+]
+keywords = [
+    "ipykernel",
+    "jupyter",
+    "xeus-python",
+]
+dependencies = [
```
Comment (martinRenou) [inline pyproject.toml:20]:
```suggestion
classifiers = [
    "Framework :: Jupyter",
    "License :: OSI Approved :: BSD License",
    "Programming Language :: Python",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.6",
    "Programming Language :: Python :: 3.7",
    "Programming Language :: Python :: 3.8",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
]
dependencies = [
```

**Pull request merged**